docs/user-guide/src/tour/pop.md

```diff
@@ -1,13 +1,172 @@
 # Pop
 
-**TODO**
+> **Recommended reading:** 
+> [2.5: Pop](https://rust-unofficial.github.io/too-many-lists/first-pop.html), 
+
+Let's continue with a function to remove and return the element at the head of a list. The way to write such a function is described in [2.5: Pop](https://rust-unofficial.github.io/too-many-lists/first-pop.html), we will focus on the verification in this chapter.
+
+There is one change to the code from the original tutorial:
+We will rename the `pop` function to `try_pop`. The return type is still `Option<i32>` and `try_pop` will return `Some(item)` if the list has elements, and `None` otherwise. We will then add a new `pop` function, which has the return type `i32`, and will panic if it is called with an empty list. However, by using the correct precondition, we can prevent the `pop` function from ever panicking! Here is the code:
+
+```rust,noplaypen
+{{#rustdoc_include ../../../../prusti-tests/tests/verify/fail/user-guide/pop.rs:initial}}
+```
+
+For the implementation of our `pop` method, we can reuse the implementation of `try_pop`. We call `try_pop` on the list, then call [`unwrap`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap) on the result. `unwrap` will return the inner value of the `Option` if it is `Some`, and will panic otherwise.
+Normally, it is bad form to use `unwrap` in production code, where you should handle potential errors, instead of just panicking.
+But, since we are verifying that there will never be `None` passed to `unwrap`, we should be able to get away with it here.
```

Comment:
(I wonder if this point is made anywhere yet:) we are indeed verifying the function and it will not panic *if* the precondition holds. When verifying an application, this is fine. However, if we are verifying library code, then the code using the library must also be verified by Prusti! Otherwise, nothing enforces that the user code is using the library correctly, and it could e.g. contain a call to `pop` over an empty list.